pubsub-mapped-api/src/test/java/com/google/pubsub/clients/consumer/KafkaConsumerTest.java

```diff
@@ -0,0 +1,333 @@
+        .put("value.deserializer",
+            "com.google.pubsub.kafkastubs.common.serialization.StringDeserializer")
+        .put("max.poll.records", 500)
+        .build()
+    );
+
+    return new ConsumerConfig(
+        ConsumerConfig.addDeserializerToConfig(properties, null, null));
+  }
+
+  static class SubscriberImpl extends SubscriberImplBase {
+
+    @Override
+    public void createSubscription(Subscription request, StreamObserver<Subscription> responseObserver) {
+      Subscription s = Subscription.newBuilder().setName("name")
+          .setTopic("projects/null/topics/topic").build();
+      responseObserver.onNext(s);
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void deleteSubscription(DeleteSubscriptionRequest request, StreamObserver<Empty> responseObserver) {
+      responseObserver.onNext(Empty.getDefaultInstance());
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void pull(PullRequest request, StreamObserver<PullResponse> responseObserver) {
+      String topic = "topic";
+      Integer key = 125;
+      String value = "value";
+      byte[] serializedKeyBytes = new IntegerSerializer().serialize(topic, key);
+      String serializedKey = new String(serializedKeyBytes, StandardCharsets.UTF_8);
+      byte[] serializedValueBytes = new StringSerializer().serialize(topic, value);
+
+      Timestamp timestamp = Timestamp.newBuilder().setSeconds(1500).build();
+
+      PubsubMessage pubsubMessage = PubsubMessage.newBuilder()
+          .setPublishTime(timestamp)
+          .putAttributes("key_attribute", serializedKey)
+          .setData(ByteString.copyFrom(serializedValueBytes))
+          .build();
+
+      ReceivedMessage receivedMessage = ReceivedMessage.newBuilder()
+          .setMessage(pubsubMessage)
+          .build();
+
+      PullResponse pullResponse = PullResponse.newBuilder()
+          .addReceivedMessages(receivedMessage)
+          .build();
+
+      responseObserver.onNext(pullResponse);
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void acknowledge(AcknowledgeRequest request, StreamObserver<Empty> responseObserver) {
+      Empty empty = Empty.getDefaultInstance();
+      responseObserver.onNext(empty);
+      responseObserver.onCompleted();
+    }
+  }
+
+  static class PublisherImpl extends PublisherImplBase {
+
+    @Override
+    public void listTopics(ListTopicsRequest request, StreamObserver<ListTopicsResponse> responseObserver) {
+      List<String> topicNames = new ArrayList<>(Arrays.asList(
+        "projects/null/topics/thisis123cat", "projects/null/topics/abc12345bad",
+        "projects/null/topics/noWay1234", "projects/null/topics/funnycats000cat"));
+      List<Topic> topics = new ArrayList<>();
+
+      for (String topicName: topicNames) {
+        topics.add(Topic.newBuilder().setName(topicName).build());
+      }
+
+      ListTopicsResponse listTopicsResponse = ListTopicsResponse.newBuilder()
```

Comment:
optional: you can construct the builder without using the intermediate arraylist.  You can call `allTopicBuilder().setName(...)` in the loop.